compare/meson.build

```diff
@@ -0,0 +1,19 @@
+compare_src = files(
+    'compare.c',
+    'image.c',
+)
+
+libass_compare = executable(
+    'compare',
+    compare_src,
+    install: false,
+    include_directories: incs,
+    dependencies: deps + png_deps,
+    link_with: libass_for_tools,
+)
+
+art_samples = get_option('art-samples')
+if art_samples != ''
+    dir = join_paths(art_samples, 'regression')
+    test('art-compare', files(join_paths(dir, 'run-all.sh'))[0], workdir: dir, args: [libass_compare.full_path()], depends: [libass_compare])
```

Comment:
How does this work anyway? AFAICT the shell is set to PowerShell, but the executable is a Bourne shell script. Is MSYS2 automatically registered as a handler for `sh` files?

Reply:
Meson runs `sh` with `run-all.sh` as the first arg.

Reply:
Interesting.

[This doc page](https://mesonbuild.com/External-commands.html) (while not specifically about `test`) says “Meson will autodetect scripts with a shebang line and run them with the executable/interpreter specified in it both on Windows and on Unixes”. I’m gonna assume tests do the same, and that on Windows it discards the explicit path from the shebang and tries to run the executable by basename from PATH etc.